File: install-page/initialize.py
"""Initialize empyrean: download SPICE kernels and load the data dir.

Reproducer for the "Initialize & Data" section of
https://empyrean-dynamics.com/install. The marked region below is the
literal snippet shown in the install page; the surrounding harness is
what makes the file `python install-page/initialize.py`-runnable.

Run:
    pip install empyrean
    python install-page/initialize.py
"""

from __future__ import annotations


def main() -> None:
    # spielberg:snippet:start
    import empyrean
    from empyrean import Epochs, TimeScale, Frame, Origin

    empyrean.download_data()  # first-run only
    empyrean.initialize()
    # spielberg:snippet:end
    _ = (Epochs, TimeScale, Frame, Origin)  # silence "unused"
    print("empyrean initialized OK")


if __name__ == "__main__":
    main()

File: install-page/initialize.rs
//! Initialize empyrean: load SPICE kernels via the Context.
//!
//! Reproducer for the "Initialize & Data" section of
//! <https://empyrean-dynamics.com/install>. The marked region below is
//! the literal snippet shown in the install page; the surrounding
//! `fn main` is what makes the file runnable as a standalone binary.
//!
//! Run:
//! ```bash
//! cargo run --release --bin install-initialize
//! ```

fn main() -> empyrean::Result<()> {
    // empyrean:snippet:start
    use empyrean::{Context, Epoch, Frame, Origin};

    // Loads / downloads kernels on first run; reads cached files thereafter.
    // Override the data directory with the EMPYREAN_DATA_DIR env var.
    let ctx = Context::from_data_dir(None)?;
    // empyrean:snippet:end
    let _ = (&ctx, Epoch::from_mjd_tdb(60000.0), Frame::ICRF, Origin::SSB);
    println!("empyrean Context initialized OK");
    Ok(())
}
